client/packages/host/src/components/AppDrawer/SyncNavLink.tsx

```diff
@@ -46,7 +60,7 @@ export const SyncNavLink = () => {
       }}
       icon={<RadioIcon fontSize="small" color="primary" />}
       text={t('sync')}
-      badgeProps={badgeProps}
+      badgeProps={badgeProps || undefined}
```

Comment:
I find typescript... obtuse. Can we just have `falsy` as a type? Lol

I think all the types for `syncStatus` (like everywhere else) a cumbersome too... I think because the type is anything that the API MIGHT return, vs specifically what we're requesting the in gql in this component. If there is a way to just have a type that matches the gql request we're making and cleans up everything MAYBE being undefined | null that'd allow us to have much nicer code while still largely being typesafe?

Reply:
Yeh... here's a suggestion - moving the getBadge to separate function, handling the non-existence of the syncInfo data, and simplifying the showWarning/showError logic a bit, date utils helps:

https://github.com/msupply-foundation/open-msupply/compare/6446-sync-badge-only-on-warning-or-error...6446-sync-badge-only-on-warning-or-error(refactor-getbadge)